pkg/operator/ceph/cluster/osd/provision_spec.go

```diff
@@ -355,3 +354,61 @@ func (c *Cluster) provisionOSDContainer(osdProps osdProperties, copyBinariesMoun
 
 	return osdProvisionContainer, nil
 }
+
+// deleteAllOrphanedPrepareJobs deletes all osd prepare jobs which are bound to kubernetes nodes which no longer exist.
+func (c *Cluster) deleteAllOrphanedPrepareJobs() {
+	listOpts := metav1.ListOptions{
+		LabelSelector: prepareJobSelector(),
+	}
+	jobClientset := c.context.Clientset.BatchV1().Jobs(c.clusterInfo.Namespace)
+	jobs, err := jobClientset.List(c.clusterInfo.Context, listOpts)
+	if err != nil {
+		logger.Warningf("failed to clean up any orphaned OSD prepare jobs. failed to list OSD prepare jobs. %v", err)
+		return
+	}
+
+	nodes, err := c.context.Clientset.CoreV1().Nodes().List(c.clusterInfo.Context, metav1.ListOptions{})
+	if err != nil {
+		logger.Warningf("failed to clean up any orphaned OSD prepare jobs. failed to list nodes. %v", err)
+		return
+	}
+
+	hostNames := set.New[string]()
+	for _, node := range nodes.Items {
+		hostName, hasHostName := node.Labels[k8sutil.LabelHostname()]
+		if hasHostName {
+			hostNames.Insert(hostName)
+		}
+	}
+
+	for _, job := range jobs.Items {
+		nodeSelector := job.Spec.Template.Spec.NodeSelector
+		jobHostName, hasHostNameSelector := nodeSelector[k8sutil.LabelHostname()]
+		if hasHostNameSelector {
+			if !hostNames.Has(jobHostName) {
+				logger.Debugf("cleaning up orphaned OSD prepare job %q.", job.Name)
+				err := jobClientset.Delete(c.clusterInfo.Context, job.Name, metav1.DeleteOptions{})
+				if err != nil {
+					logger.Warningf("failed to clean up OSD prepare job %q. %v", job.Name, err)
+				}
+			}
+		}
+	}
+}
+
+func prepareJobSelector() string {
+	return fmt.Sprintf("%s=%s",
```

Comment:
nit: I'd suggest just to put this inline instead of creating a helper method, since it's only called in one place.